src/coord/rmr/command.c

```diff
@@ -159,14 +168,11 @@ void MRCommand_SetPrefix(MRCommand *cmd, const char *newPrefix) {
   MRCommand_ReplaceArgNoDup(cmd, 0, buf, len);
 }
 
-void MRCommand_ReplaceArgNoDup(MRCommand *cmd, int index, const char *newArg, size_t len) {
-  if (index < 0 || index >= cmd->num) {
-    return;
-  }
-  char *tmp = cmd->strs[index];
-  cmd->strs[index] = (char *)newArg;
+void MRCommand_ReplaceArgNoDup(MRCommand *cmd, int index, char *newArg, size_t len) {
+  RS_ASSERT(0 <= index && index < cmd->num);
```

Comment:
Would `RS_ASSERT_ALWAYS` + log be better here?

Reply:
1. Asserts always come with a log, in this case it's self explanatory so I didn't add text.
2. We can `RS_ASSERT_ALWAYS`, but I think we should always have a good justification for the index when using any index-based operation on the command, so asserting on the CI and should be enough